File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        
        int n = cardPoints.length;
        
        int lSum = 0;
        int rSum = 0;
        for(int i=n-1; i>=n-k; i--){  // taking sum of last k cards
            rSum += cardPoints[i];
        }
        
        
        int maxPoints = rSum;
        int i=0, j=(n-k);
        for(int l=0; l<k; l++){
            int points = 0;
           
            lSum += cardPoints[i++];
            rSum -= cardPoints[j++];
            
            points = lSum+rSum;
            System.out.println(points);
            maxPoints = Math.max(points, maxPoints);
        }
        
        // maxPoints = Math.max(lSum, maxPoints);
        
        return maxPoints;
    }
}

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:
    int maxScore(vector<int>& num, int k) {
        int n = num.size();
        
        int lSum = 0;
        int rSum = 0;
        for(int i=n-1; i>=n-k; i--){
            rSum += num[i];
        }
        // System.out.println(rSum);
        
        int left = 0;
        int right = k;
        
        int maxScore = rSum;
        int i=0, j=(n-k);
        while(left <= k-1 && right >= 1){
            int sum = 0;
           
            lSum += num[i++];
            rSum -= num[j++];
            
            sum = lSum+rSum;
            
            maxScore = max(sum, maxScore);
            left++;
            right--;
        }
        
        maxScore = max(lSum, maxScore);
        
        return maxScore;
    }
};